ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -383,11 +384,12 @@ void checkMaintainedConnectionPeers() {
   @VisibleForTesting
   void attemptPeerConnections() {
     LOG.trace("Initiating connections to discovered peers.");
-    rlpxAgent.connect(
+    final Stream<DiscoveryPeer> toTry =
         streamDiscoveredPeers()
             .filter(peer -> peer.getStatus() == PeerDiscoveryStatus.BONDED)
             .filter(peerDiscoveryAgent::checkForkId)
-            .sorted(Comparator.comparing(DiscoveryPeer::getLastAttemptedConnection)));
+            .sorted(Comparator.comparing(DiscoveryPeer::getLastAttemptedConnection));
+    toTry.forEach(rlpxAgent::connect);
```

Comment:
If I understood this correctly, this is called every 30 seconds, how many peers can be attempted every time? 
Asking to see if we are doing too many attempts, and if there is a need to put some limits, so we try at max tot connections at time.

Reply:
This PR is reducing the number of peers we are trying to connect to to the ones that have a valid fork id or did not send a fork id. 
I agree that we should look into limiting this, e.g. if we do have reached max peers. But that is for another time :-)

Reply:
I will add a new task to make sure that we can have a look at optimising that!